packages/element/src/raw-html.js

```diff
@@ -3,17 +3,21 @@
  */
 import { createElement } from './react';
 
+// Disable reason: JSDoc linter doesn't seem to parse the union (`&`) correctly.
+/* eslint-disable jsdoc/valid-types */
+/** @typedef {{children: string} & import('react').ComponentPropsWithoutRef<'div'>} RawHTMLProps */
+/* eslint-enable jsdoc/valid-types */
+
 /**
  * Component used as equivalent of Fragment with unescaped HTML, in cases where
  * it is desirable to render dangerous HTML without needing a wrapper element.
  * To preserve additional props, a `div` wrapper _will_ be created if any props
  * aside from `children` are passed.
  *
- * @param {Object} props
- * @param {string} props.children HTML to render.
- * @param {Object} props.props    Any additonal props to be set on the containing div.
+ * @param {RawHTMLProps} props Children should be a string of HTML. Other props
+ *                             will be passed through to div wrapper.
  *
- * @return {WPComponent} Dangerously-rendering component.
+ * @return {JSX.Element} Dangerously-rendering component.
```

Comment:
I wondered how this worked.

Turns out it doesn't:

```
[0] /home/travis/build/WordPress/gutenberg/packages/element/src/raw-html.js
[0]   20:0  warning  The type 'JSX' is undefined  jsdoc/no-undefined-types
```

https://travis-ci.com/github/WordPress/gutenberg/jobs/314225929

I thought these rules were upgraded to be errors in #20427. I think this one might have been missed? (cc @sainthkh)

I know we have a handful of temporary exceptions for custom types, but pretty sure those are hard-coded to be exceptions and could still exist even if the rule were upgraded from warning to error.

Reply:
Is this a concern with restoring the element types? #21781

Reply:
It happened because I missed this one rule: 

https://github.com/WordPress/gutenberg/blob/5bd92c649583511151b56b13664f2807579457ba/packages/eslint-plugin/configs/jsdoc.js#L82-L83

I opened an issue that fixes it: #21942.